lib/src/realm_class.dart

```diff
@@ -132,29 +128,22 @@ class Realm implements Finalizable {
   /// Returns [Future<Realm>] that completes with the `realm` once the remote realm is fully synchronized or with an `error` if operation is canceled.
   /// When the configuration is [LocalConfiguration] this completes right after the local realm is opened or operation is canceled.
   static Future<Realm> open(Configuration config, {CancellationToken? cancellationToken, ProgressCallback? onProgressCallback}) async {
-    Realm realm = await CancellableFuture.fromFutureFunction<Realm>(() => _open(config, cancellationToken, onProgressCallback), cancellationToken);
-    return realm;
+    final realm = Realm(config);
```

Comment:
I wanted to prevent this Realm to be opened before the cancellation logic to be attached. If I call cancellationToken.cancel() before to attach the releasing event that closes the Realm, then the Realm is opened successfully whiteout to be closed and the cancelled exception is thrown later.
And because of the exception the users can't get access to this realm instance and can never close it.

Reply:
GC will close it. Realms are FInalizable, so it will be prioritezed

Reply:
But the tests fail on teardown. :) And I prefer to be closed, not to leave any garbage to collect :)

Reply:
No it does not fail on teardown.